src/service/item.service.ts

```
@@ -33,42 +49,42 @@ const add = async (newItem: FEItemInterface): Promise<ItemInterface> => {
     throw new Error(err);
   }
 };
+const remove = async (id: string): Promise<Boolean> => {
+  try {
+    await itemModel.findByIdAndUpdate(id, { isActive: false });
+    return true;
+  } catch (err: any) {
+    throw new Error(err);
+  }
+};
 
-const remove = async(id : string) : Promise<Boolean> =>{
-    try{
-        await itemModel.findByIdAndUpdate(id, {isActive: false})
-        return true;
-    }
-    catch (err: any) {
-        throw new Error(err)
+const edit = async (
+  item: FEItemInterface,
+  id: string
+): Promise<FEItemInterface> => {
+  try {
+    const itemToUpdate = await itemModel.findById(id);
+    if (!itemToUpdate) {
```

you also need to check if `!itemToUpdate.isActive`, then you want to throw the item not found error, cause the item has been deleted